libpromises/locks.c

```diff
@@ -889,7 +889,7 @@ void GetLockName(char *lockname, const char *locktype, const char *base, const R
     strncpy(lockname, locktype, CF_BUFSIZE / 10);
     strcat(lockname, "_");
     strncat(lockname, base, CF_BUFSIZE / 10);
-    strcat(lockname, "_");
+    strlcat(lockname, "_", CF_BUFSIZE / 10);
```

Comment:
Oh, and while you're at it, switch them all to `l` version, which is better than `n` version. I forget the exact details, but the `n` version doesn't include the null byte in the calculation or something along those lines.